src/ORM/Rule/ExistsIn.php

```diff
@@ -129,4 +146,28 @@ protected function _fieldsAreNull($entity, $source)
         }
         return $nulls === count($this->_fields);
     }
+
+    /**
+     * Checks whether or not the give entity fields are null and map to schema NULL
+     * or are not null and map to schema NOT NULL.
+     *
+     * @param \Cake\Datasource\EntityInterface $entity The entity to check.
+     * @param \Cake\ORM\Table $source The table to use schema from.
+     * @return bool
+     */
+    protected function _checkPartialSchemaNulls($entity, $source)
+    {
+        $schema = $source->schema();
+        foreach ($this->_fields as $field) {
+            $isNullable = $schema->isNullable($field);
+            $value = $entity->get($field);
+            if (!$isNullable && $value === null) {
+                return false;
+            }
+            if ($isNullable && $value !== null) {
+                return false;
```

Comment:
If a column is allowed to be SQL `NULL` but is not set to php `null` it should be checked just like any other field:

See: https://github.com/ionas/cakephp/blob/1ec0ab73c0e9dd10893b38a374717fba5e2c8ef1/src/ORM/Rule/ExistsIn.php#L116-L128 especially: https://github.com/ionas/cakephp/blob/1ec0ab73c0e9dd10893b38a374717fba5e2c8ef1/src/ORM/Rule/ExistsIn.php#L128

Does that make sense to you?